addons/sale_timesheet/models/sale_order.py

```diff
@@ -67,6 +67,7 @@ def action_view_task(self):
         action = {'type': 'ir.actions.act_window_close'}
 
         task_projects = self.tasks_ids.mapped('project_id')
+        self = self.with_context(active_id=task_projects.id)
```

Comment:
Any chances of Expected Singleton here ?

Reply:
@sswapnesh thanks for your feedback, I will do some further testing before submitting a new version